plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -307,12 +309,22 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
         if (type instanceof ArrayType || type instanceof MapType || type instanceof RowType) {
             // TODO https://github.com/trinodb/trino/issues/17583 Add support for these complex types
             throw new TrinoException(NOT_SUPPORTED, "Migrating %s type is not supported".formatted(type));
         }
+
+        if (type.equals(TimestampType.TIMESTAMP_MICROS)) {
+            // TODO (https://github.com/trinodb/trino/pull/17221) Support for Avro
+            return switch (storageFormat) {
+                case ORC, PARQUET -> Types.TimestampType.withoutZone();
+                // TODO (https://github.com/trinodb/trino/pull/17221) Support for Avro
+                case AVRO -> throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp(3) type with %s format is not supported.".formatted(storageFormat));
+                default -> throw new TrinoException(NOT_SUPPORTED, "Migration data in %s format is not supported. Use ORC or Parquet instead.".formatted(storageFormat));
+            };
+        }
```

Comment:
Rejecting unsupported table types should be done regardless of timestamp type.
Indeed it is, in `toIcebergFileFormat(HiveStorageFormat)`, so this would suffice

```suggestion
if (type instanceof TimestampType && storageFormat == AVRO) {
            throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp(3) type with %s format is not supported.".formatted(storageFormat));
        }
```

(note that it would keep delegating to `toIcebergTypeForNewColumn` for supported cases)